pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerViewModel.kt

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * Simple player view model than handle a PillarboxPlayer [player]
+ */
+class SimplePlayerViewModel(application: Application) : AndroidViewModel(application), Player.Listener {
+    /**
+     * Player as PillarboxPlayer
+     */
+    val player = PillarboxPlayer(application, DemoMediaItemSource(DemoItemDataSource(application)))
+
+    init {
+        player.addListener(this)
+    }
+
+    /**
+     * Add to [player] all [ids] to the MediaItem list.
+     * Will prepare and play the content.
+     *
+     * @param ids mediaIdentifier to play
+     */
+    fun playItemIds(ids: Array<String>) {
+        player.addMediaItems(ids.map { fromMediaId(it) })
+        player.prepare()
+        player.play()
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        player.release()
+        player.removeListener(this)
+    }
+
+    /**
+     * Resume playback of [player]
+     */
+    fun resumePlayback() {
+        player.play()
+    }
+
+    /**
+     * Pause playback of [player]
+     */
+    fun pausePlayback() {
+        player.pause()
+    }
+
+    override fun onMediaMetadataChanged(mediaMetadata: MediaMetadata) {
+        Log.d(TAG, "onMediaMetadataChanged title = ${mediaMetadata.title}")
+    }
+
+    override fun onPlaylistMetadataChanged(mediaMetadata: MediaMetadata) {
+        Log.d(TAG, "onPlaylistMetadataChanged title = ${mediaMetadata.title}")
+    }
+
+    override fun onTimelineChanged(timeline: Timeline, reason: Int) {
+        val reasonString = when (reason) {
+            Player.TIMELINE_CHANGE_REASON_PLAYLIST_CHANGED -> "TIMELINE_CHANGE_REASON_PLAYLIST_CHANGED"
+            Player.TIMELINE_CHANGE_REASON_SOURCE_UPDATE -> "TIMELINE_CHANGE_REASON_SOURCE_UPDATE"
+            else -> "?"
+        }
+        Log.d(
+            TAG,
+            "onTimelineChanged $reasonString ${player.currentMediaItem?.mediaId}" +
+                " ${player.currentMediaItem?.mediaMetadata?.title}" +
+                " uri = ${player.currentMediaItem?.localConfiguration?.uri}" +
+                " tag=${player.currentMediaItem?.localConfiguration?.tag}"
+        )
+    }
+
+    override fun onPlayWhenReadyChanged(playWhenReady: Boolean, reason: Int) {
+        val reasonString = when (reason) {
+            Player.PLAY_WHEN_READY_CHANGE_REASON_AUDIO_BECOMING_NOISY -> "PLAY_WHEN_READY_CHANGE_REASON_AUDIO_BECOMING_NOISY"
+            Player.PLAY_WHEN_READY_CHANGE_REASON_AUDIO_FOCUS_LOSS -> "PLAY_WHEN_READY_CHANGE_REASON_AUDIO_FOCUS_LOSS"
+            Player.PLAY_WHEN_READY_CHANGE_REASON_REMOTE -> "PLAY_WHEN_READY_CHANGE_REASON_REMOTE"
+            Player.PLAY_WHEN_READY_CHANGE_REASON_USER_REQUEST -> "PLAY_WHEN_READY_CHANGE_REASON_USER_REQUEST"
+            Player.PLAY_WHEN_READY_CHANGE_REASON_END_OF_MEDIA_ITEM -> "PLAY_WHEN_READY_CHANGE_REASON_END_OF_MEDIA_ITEM"
+            else -> "?"
+        }
+        Log.d(TAG, "onPlayWhenReadyChanged $reasonString ${player.currentMediaItem?.mediaMetadata?.title}")
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SRGSSR/pillarbox-android/blob/9e672e88664d24e42b2fbb65a69ff4507dff46b6/pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/SimplePlayerViewModel.kt#L97"></span>The log call Log.d(…) should be conditional: surround with <code>if (Log.isLoggable(...))</code> or <code>if (BuildConfig.DEBUG) { ... }</code></td>
    </tr>
  </tbody>
</table>